### ISSUE TITLE
Heavy use of mutable globals in Go code

### ISSUE THREAD

**Issue by kpenfound** [nanu-c/axolotl#828]
The Go code under `/app` uses mutable globals which makes state hard to debug, creates opportunities for race conditions, and makes unit testing difficult.

Some examples of what I'm referring to are:
- config https://github.com/nanu-c/axolotl/blob/main/app/config/config.go#L26
- settings https://github.com/nanu-c/axolotl/blob/main/app/settings/settings.go#L22
- worker https://github.com/nanu-c/axolotl/blob/main/app/worker/textsecureapi.go#L33-L36

**Describe the solution you'd like**
Mutable globals should be removed entirely.

At a high level, a struct managing state should be passed through to different modules on initialization. Modules which vendor globals should instead have these values as properties of their own struct, which will be a component of the top level struct managing the state.

For example, under `/app` we will have an entrypoint for the app as a whole, defined as
```
type App struct {
    Config   *config.Config
    Settings *settings.Settings
    ...
}
```
where `app.Settings` would operate the same as the global `SettingsModel` today, but passed as a pointer instead of a global.

Some globals, particularly in config, can be computed based on context and don't necessarily need to be passed as app state. These can be provided as getters on the config module rather than as a global.

**Comment by kpenfound** [comment]
To clarify the scope on this, here's my first commit working through this https://github.com/kpenfound/axolotl/commit/33739a7aae08c53e3e63329a6045e5ff608dcfe9

Most of the changes for config and settings are included here, but there's still a lot to be done
The globals still in use are marked as `// TODO`

**Comment by nanu-c** [comment]
Hey @kpenfound thank you for your engagement and i really like your approach. That commit changes a lot of things and we haven't defined any coding styles anywhere in this project but i like to add them before we make such huge changes. Maybe we can discuss them here.

- Don't use abbreviations for variable names.
  - `app := setup()` instead of `a := setup()`
  - `func runBackend(appplication *app.App) {` instead of `func runBackend(a *app.App) {`
  - `registeredContactsFile := config.GetRegisteredContactsFile()` instead of `rcf := config.GetRegisteredContactsFile()`
  - Only exception are functions that are extending `structs` like `func (c *Config) Unregister() {`. There should be used the first letter in lowercase.
 - Don't pass complex data types like `structs` through multiple functions.

 ``` 
func MessageHandler(msg *textsecure.Message, config *config.Config) {
	buildAndSaveMessage(msg, false, config)
}
```
should be (or similiar, this is pseudo code)
 ``` 
type handler struct {
   Config *cofig.Config
}
func (h *Handler) MessageHandler(msg *textsecure.Message) {
   h.Config = config.GetConfig()
   h.BuildAndSaveMessage(msg, false)
}
```
- avoid mutable global functions outside of structs. 
 - `func (c *Config)GetIsPhone() bool{}` instead of `func GetIsPhone() bool {}`
 - attachDir := config.GetConfig().GetAttachDir()
- consistent naming of variables that are the same in multiple functions
 - ex. usage of `cf := config.GetContactsFile()` and `contactsFile := config.GetContactsFile()`. It should be the same in both cases
- Don't pass variables as function parameters that belong to the config
```
func TelToName(tel string, ourTel string) string {
        [...]
	if tel == ourTel {
		return "Me"
	}
}
```
should be
```
func TelToName(tel string) string {
        [...]
	if tel == config.GetConfig().GetMyNumber() {
		return "Me"
	}
}
```

I think that's a good starting point. Some of these points are automatically checked by our github actions when you open a pull request. So feel free to do so as a WIP branch. 
Thank you 👍🏾

**Comment by kpenfound** [comment]
Thanks for the detailed write-up @nanu-c , that all sounds like a great style to conform to. I'll do some tidying up and get a draft PR going, and I'll try to quantify the progress there.

**Comment by kpenfound** [comment]
@nanu-c thoughts on removing unused functions over refactoring them? I've run into few that are no longer used and have refactored them so far, but the more I run into the more I think removal may be better

**Comment by Blackoverflow** [comment]
Another coding convention I'd like would be to isolate all side effects through interfaces, in order to be able to mock them easily.

```golang
interface messageStore {
  save(messageEntity MessageEntity)
}
type handler struct {
  MessageStore *MessageStore
}

func (h *Handler) buildAndSaveMessage(msg *textsecure.Message, outgoing ) {
   messageEntity := buildMessage(msg, outgoing)
   MessageStore,save(messageEntity)
}
```

**Comment by nanu-c** [comment]
will be solved with #985
